### PR TITLE
Implement exception handling for put request

### PR DESCRIPTION
**Pull request by sitaram-kalluri**
**- What I did**
Add exception chaining to put method in AtClient Impl

**- How I did it**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: at_client/lib/src/client/at_client_impl.dart [hunk 1]
```diff
@@ -361,45 +361,51 @@ class AtClientImpl implements AtClient {
...
+        ..one = atKey
+        ..two = value;
+      // Transform put request
+      UpdateVerbBuilder verbBuilder =
+          await PutRequestTransformer().transform(tuple);
+      // Execute the verb builder
+      var putResponse = await SecondaryManager.getSecondary(verbBuilder)
+          .executeVerb(verbBuilder, sync: SyncUtil.shouldSync(atKey.key!));
+      // If putResponse is null or empty, return empty response
+      if (putResponse == null || putResponse.isEmpty) {
```
Comment (murali-shris) [inline at_client/lib/src/client/at_client_impl.dart]: should this check be in put transfomer?

File: at_client/lib/src/client/remote_secondary.dart [hunk 2]
```diff
@@ -56,8 +56,10 @@ class RemoteSecondary implements Secondary {
     try {
       verbResult = await executeVerb(builder);
       verbResult = verbResult.replaceFirst('data:', '');
-    } on AtClientException catch (e) {
-      logger.severe('Exception occurred in processing the verb ${e.message}');
+    }on AtException catch (e) {
```
Comment (murali-shris) [inline at_client/lib/src/client/remote_secondary.dart:59]: formatting issue

**Review by gkc — CHANGES_REQUESTED** [review]
LGTM. Will not approve yet as we need to not merge any new changes into trunk until https://github.com/atsign-foundation/at_client_sdk/pull/514 has been fully tested, merged into trunk and new version of at_client package published

**Review by murali-shris — APPROVED** [review]

**Comment by sitaram-kalluri** [comment]
> LGTM. Will not approve yet as we need to not merge any new changes into trunk until #514 has been fully tested, merged into trunk and new version of at_client package published

@gkc : We have published the at_client version which has PR 514. Can you please review the PR once again.

**Review by gkc — APPROVED** [review]

**Pull request merged**